libservices/CMakeLists.txt

```diff
@@ -42,3 +42,42 @@ SET(MYSQLSERVICES_SOURCES
 
 ADD_CONVENIENCE_LIBRARY(mysqlservices ${MYSQLSERVICES_SOURCES})
 INSTALL(TARGETS mysqlservices DESTINATION ${INSTALL_LIBDIR} COMPONENT Development)
+
+MACRO(CREATE_AIX_EXPORT_FILE op outfile version symbols alias_version)
+  SET(EXPORT_CONTENT "#! .\n")
+  FOREACH(exp_symbol ${symbols})
+    SET(EXPORT_CONTENT ${EXPORT_CONTENT} "${exp_symbol}\n")
+  ENDFOREACH()
+  FILE(${op} ${CMAKE_CURRENT_BINARY_DIR}/${outfile} ${EXPORT_CONTENT})
+ENDMACRO()
+
+IF(CMAKE_SYSTEM_NAME MATCHES AIX)
+  SET(MYSQLSERVICES_SYMBOLS
+    base64_service
+    debug_sync_service
+    encryption_scheme_service
+    encryption_service
+    json_service
+    logger_service
+    my_crypt_service
+    my_md5_service
+    my_print_error_service
+    my_sha1_service
+    my_sha2_service
+    my_snprintf_service
+    progress_report_service
+    thd_alloc_service
+    thd_autoinc_service
+    thd_error_context_service
+    thd_kill_statement_service
+    thd_rnd_service
+    thd_specifics_service
+    thd_timezone_service
+    thd_wait_service
+    wsrep_service
+  )
+  CREATE_AIX_EXPORT_FILE(WRITE mysqlservices.def
```

Comment:
I think you can just check in the mysqlservices_aix.def, if you do not find any other possibility to make all symbols in mysqlservices visible, e.g via visibility attribute, or -fvisibility or -qvisibility.

Anyway,  no need to generate it, every time cmake is invoked. BTW, if you want to generate some text file, used in build, prefer CONFIGURE_FILE to FILE(WRITE), as CONFIGURE_FILE does not change timestamps, causing less rebuilds.

Reply:
Done.